src/backend/commands/publicationcmds.c

```diff
@@ -573,6 +573,12 @@ RemovePublicationById(Oid pubid)
 	if (pubform->puballtables)
 		CacheInvalidateRelcacheAll();
 
+	/*
+	 * CBDB GITHUB ISSUE:
+	 * https://github.com/cloudberrydb/cloudberrydb/issues/504
+	 */
+	MetaTrackDropObject(PublicationRelationId, pubid);
+
```

Comment:
We'd better not mix these lines with the surrounding code since they are unrelated.